core/templates/dev/head/exploration_editor/ExplorationEditor.js

```diff
@@ -620,49 +624,140 @@ oppia.controller('ExplorationSaveAndPublishButtons', [
-        explorationData.explorationId);
-      alertsService.clearWarnings();
-
-      // If the objective has not yet been specified, open the pre-publication
-      // 'add exploration metadata' modal.
-      if (!explorationObjectiveService.savedMemento) {
-        $modal.open({
-          templateUrl: 'modals/addExplorationMetadata',
-          backdrop: true,
-          controller: [
-            '$scope', '$modalInstance', 'explorationObjectiveService',
-            function($scope, $modalInstance, explorationObjectiveService) {
-              // If no objective has been specified yet, require the creator
-              // to specify it.
-              $scope.explorationObjectiveService = explorationObjectiveService;
-              $scope.requireObjectiveToBeSpecified = (
-                !explorationObjectiveService.savedMemento);
-
-              $scope.save = function() {
-                if ($scope.requireObjectiveToBeSpecified) {
-                  if (!explorationObjectiveService.displayed) {
-                    throw Error('Please specify an objective');
-                  }
-                  explorationObjectiveService.saveDisplayedValue();
-                }
-                $modalInstance.close();
-              };
-
-              $scope.cancel = function() {
-                $modalInstance.dismiss('cancel');
-                alertsService.clearWarnings();
-              };
-            }
-          ]
-        }).result.then(function() {
-          saveDraftToBackend('Add an objective.', openPublishExplorationModal);
-        });
-      } else {
-        // No further metadata is needed. Open the publish modal immediately.
-        openPublishExplorationModal();
-      }
-    };
+       siteAnalyticsService.registerOpenPublishExplorationModalEvent(
+         explorationData.explorationId);
+       alertsService.clearWarnings();
+
+       // If the metedata has not yet been specified, open the pre-publication
+       // 'add exploration metadata' modal.
+       if (!explorationTitleService.savedMemento ||
+          !explorationObjectiveService.savedMemento ||
+          !explorationCategoryService.savedMemento ||
+          !explorationLanguageCodeService.savedMemento) {
+         $modal.open({
+           templateUrl: 'modals/addExplorationMetadata',
+           backdrop: true,
+           controller: [
+             '$scope', '$modalInstance', 'explorationObjectiveService',
+             'explorationTitleService', 'explorationCategoryService',
+             'explorationStatesService', 'CATEGORY_LIST',
+             'explorationLanguageCodeService',
+             function($scope, $modalInstance, explorationObjectiveService,
+             explorationTitleService, explorationCategoryService,
+             explorationStatesService, CATEGORY_LIST,
+             explorationLanguageCodeService) {
+               $scope.explorationTitleService =
+               explorationTitleService;
```

Comment:
So most of these can fit onto the previous line BUT the linter said that $scope.explorationLanguageCodeService = explorationLanguageCodeService; was too long so I was figuring that they should all look the same. Should I only change this last one and put the others back the way they were?